model/plotWidget.cc

```diff
@@ -372,7 +372,7 @@ namespace minsky
   void PlotWidget::addPlotPt(double t)
   {
     for (size_t pen=0; pen<2*numLines; ++pen)
-      if (yvars[pen].numElements()==1 && yvars[pen].idx()>=0)
+      if ((yvars[pen].numDenseElements()==1 || yvars[pen].numSparseElements==1) && yvars[pen].idx()>=0)
```

Comment:
In this client of VariableValue's code, we should be unaware of whether the storage is dense or sparse. In this case, a method that returns the actual number of elements suffices.

Reply:
Done, thanks for the suggestion.